metrics/lib/common.bash

```diff
@@ -42,6 +42,27 @@ function save_results(){
 	bash $LIB_DIR/send_results.sh -n "$1" -a "$2" -r "$3" -u "$4"
 }
 
+# This function checks existence of commands.
+# They can be received standalone standalone or as an array, e.g.
+#
+# cmds=(“cmd1” “cmd2”)
+# check_cmds "${cmd[@]}"
+function check_cmds()
+{
+	set +e
```

Comment:
The "set -e" is a global switch for the script. @bwarden is correct that the functions are not run in a subshell. The easy way to see if "set -e" is already in effect is to look at the variable $- (dollar minus) which has the currect flags. So you can do

local OLDFLAGS=$-
set +e
....
case "$OLDFLAGS" in
    (*e*) ;; # We were already in a 'set +e' state, do nothing
    (*) set -e ;;
esac